internal/controller/pkg/revision/deployment.go

```diff
@@ -66,6 +76,21 @@ func buildProviderDeployment(provider *pkgmetav1.Provider, revision v1.PackageRe
 		},
 		ImagePullSecrets: pullSecrets,
 	}
+	secSer := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            *revision.GetTLSServerSecretName(),
+			Namespace:       namespace,
+			OwnerReferences: []metav1.OwnerReference{meta.AsController(meta.TypedReferenceTo(revision, v1.ProviderRevisionGroupVersionKind))},
+		},
+	}
+
+	secCli := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            *revision.GetTLSClientSecretName(),
```

Comment:
this crashes if the secret name is not set.

Reply:
```
E0906 15:18:08.574869       1 runtime.go:79] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 406 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic({0x1883e60?, 0x2e4e750})
	k8s.io/apimachinery@v0.28.1/pkg/util/runtime/runtime.go:75 +0x7c
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile.func1()
	sigs.k8s.io/controller-runtime@v0.16.1/pkg/internal/controller/controller.go:108 +0xa4
panic({0x1883e60, 0x2e4e750})
	runtime/panic.go:884 +0x1f4
github.com/crossplane/crossplane/internal/controller/pkg/revision.buildProviderDeployment(0x40005c5600, {0x1ec2d10, 0x4000a42200}, 0x4000363680, {0x400015dc70, 0xe}, {0x0, 0x0, 0x0})
	github.com/crossplane/crossplane/internal/controller/pkg/revision/deployment.go:83 +0x394
github.com/crossplane/crossplane/internal/controller/pkg/revision.(*ProviderHooks).Post(0x40007613c0, {0x1ea4688, 0x400122e240}, {0x1e8f720, 0x40005c5600}, {0x1ec2d10?, 0x4000a42200})
	github.com/crossplane/crossplane/internal/controller/pkg/revision/hook.go:155 +0x1c8
github.com/crossplane/crossplane/internal/controller/pkg/revision.(*Reconciler).Reconcile(0x40000a9e10, {0x1ea46c0, 0x400122e1b0}, {{{0x0, 0x0}, {0x40006b3bc0, 0x33}}})
	github.com/crossplane/crossplane/internal/controller/pkg/revision/reconciler.go:675 +0x302c
github.com/crossplane/crossplane-runtime/pkg/ratelimiter.(*Reconciler).Reconcile(0x400076f630, {0x1ea46c0, 0x400122e1b0}, {{{0x0?, 0x5?}, {0x40006b3bc0?, 0x4000db5d38?}}})
	github.com/crossplane/crossplane-runtime@v1.14.0-rc.0.0.20230906075713-a2674ee167ac/pkg/ratelimiter/reconciler.go:54 +0x124
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0x1ea7a68?, {0x1ea46c0?, 0x400122e1b0?}, {{{0x0?, 0xb?}, {0x40006b3bc0?, 0x0?}}})
	sigs.k8s.io/controller-runtime@v0.16.1/pkg/internal/controller/controller.go:119 +0x8c
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0x4000765ea0, {0x1ea4618, 0x4000227400}, {0x196c540?, 0x400075f440?})
	sigs.k8s.io/controller-runtime@v0.16.1/pkg/internal/controller/controller.go:316 +0x30c
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0x4000765ea0, {0x1ea4618, 0x4000227400})
	sigs.k8s.io/controller-runtime@v0.16.1/pkg/internal/controller/controller.go:266 +0x1a0
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2()
	sigs.k8s.io/controller-runtime@v0.16.1/pkg/internal/controller/controller.go:227 +0x74
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2
	sigs.k8s.io/controller-runtime@v0.16.1/pkg/internal/controller/controller.go:223 +0x43c
```

Reply:
After discussing with @sttts, we noticed that this happens when we only change the Crossplane image without a helm upgrade. We shouldn't see this error with proper installation/upgrades. But I opened a [PR](https://github.com/crossplane/crossplane/pull/4590) to add nil checks to be on the safe side.